src/main/java/uk/gov/companieshouse/confirmationstatementapi/eligibility/impl/CompanyStatusValidation.java

```diff
@@ -0,0 +1,26 @@
+package uk.gov.companieshouse.confirmationstatementapi.eligibility.impl;
+
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+import uk.gov.companieshouse.confirmationstatementapi.eligibility.EligibilityFailureReason;
+import uk.gov.companieshouse.confirmationstatementapi.eligibility.EligibilityRule;
+import uk.gov.companieshouse.confirmationstatementapi.exception.EligibilityException;
+
+import java.util.Set;
+
+public class CompanyStatusValidation implements EligibilityRule<CompanyProfileApi> {
+
+    private final Set<String> allowedStatuses;
+
+    public CompanyStatusValidation(Set<String> allowedStatuses) {
+        this.allowedStatuses = allowedStatuses;
+    }
+
+    @Override
+    public void validate(CompanyProfileApi input) throws EligibilityException {
```

Comment:
are you able to rename input to api? just think it would read better

Reply:
shouldn't we be validating the company profile and not the api?

Reply:
ok so CompanyProfileApi is returned from the sdk but is actually the company profile